src/lib/services/group-service.ts

```diff
@@ -71,14 +84,58 @@ export class GroupService {
         return newGroup;
     }
 
-    async validateGroup({ name, users }: IGroupModel): Promise<void> {
+    async updateGroup(group: IGroupModel, userName: string): Promise<IGroup> {
+        const preData = await this.groupStore.get(group.id);
+
+        await this.validateGroup(group, preData);
+
+        const newGroup = await this.groupStore.update(group);
+
+        const existingUsers = await this.groupStore.getAllUsersByGroups([
+            group.id,
+        ]);
+        const existingUserIds = existingUsers.map((g) => g.userId);
+
+        await this.groupStore.addNewUsersToGroup(
+            newGroup.id,
+            group.users.filter(
+                (user) => !existingUserIds.includes(user.user.id),
+            ),
+            userName,
+        );
+
+        const deletableUsers = existingUsers.filter(
+            (existingUser) =>
+                !group.users.some(
+                    (groupUser) => groupUser.user.id == existingUser.userId,
+                ),
+        );
+        await this.groupStore.deleteOldUsersFromGroup(deletableUsers);
```

Comment:
There's a lot of queries going on in this function. I think mostly this is fine but I do feel that we should wrap the remove users and add new users into a transaction if possible. Probably not going to be a problem but if there ever is a race condition that leaves groups in a weird state this is going to be horrid to debug

Reply:
Agreed totally. I will do it in new pull request.